src/main/java/psidev/psi/tools/xxindex/GzXmlElementExtractor.java

```diff
@@ -39,29 +39,55 @@
 
     private boolean useSystemDefaultEncoding;
     private Charset encoding;
-
+    private FileInputStream fis;
 
     ////////////////////
     // Constructor
 
     /**
      * Default constructor setting the default character encoding to 'ASCII'.
+     * @throws IOException 
      */
-    public GzXmlElementExtractor() {
+    public GzXmlElementExtractor(File file) throws IOException {
+        
+        if (file == null) {
+            throw new IllegalArgumentException("Source file must not be null!");
+        }
+        if (!file.exists() || !file.canRead()) {
+            throw new IllegalArgumentException("Invalid source file! Can not read from: " + file.getAbsolutePath());
+        }
+        if (!file.getName().endsWith(".gz")) {
+            throw new IllegalArgumentException("Source file does not seem to be a .gz file! Consider using another XmlElementExtractor.");
+        }
+        
         setUseSystemDefaultEncoding(false);
         setEncoding(Charset.forName("ASCII"));
+        
+        // create a input stream on a gz compressed file with 1MB buffer size
+        fis = new FileInputStream(file);
+
+        
     }
 
     /**
      * Constructor overwriting the default character encoding with the specified one.
      *
      * @param encoding The Charset to use to translate the read bytes.
+     * @throws IOException 
      */
-    @SuppressWarnings(value = "unused")
-    public GzXmlElementExtractor(Charset encoding) {
-        this();
+    public GzXmlElementExtractor(File file, Charset encoding) throws IOException {
```

Comment:
@tomas-pluskal changing the constructor is a major changes for us because xxindex is used by other libraries like jmzML, jmzIdentML and others. I would take sometime to change all of this libraries.